fastparquet/util.py

```diff
@@ -53,8 +62,15 @@ def tempdir():
 
 
 def ensure_bytes(s):
+    if PY2:
+        try:
+            b = bytes(s)
+        except UnicodeEncodeError as e:
+            u = u''.join((s)).encode('utf-8').strip()
```

Comment:
Er, what is the reasoning behind this line?
Let me suggest a different implementation:
```
if PY2:
    def ensure_bytes(s):
        return s.encode('utf-8') if isinstance(s, unicode) else s
else:
    def ensure_bytes(s):
        return s.encode('utf-8') if isinstance(s, str) else s
```